### ISSUE TITLE
Connection test doesn't use https even when specifying https

### ISSUE THREAD

**Issue by rklasen** [wallabag/android-app#700]
Hi,

so this issue refernces addresses the famous "wallabag service not found" error. I just installed the android app and pointed it to my instance, but it failed to connect.

I'm using a fresh docker instance and apache to reverse proxy (I know, I know). Since I use acme.sh to get Let's Encrypt certificates, all calls to http://*.mydomain.tld point to the same static html. That means "http**s**://wallabag.mydomain.tld" points to the real Wallabag instance and "http://wallabag.mydomain.tld" points to a static html file which of course can't be parsed by the app.

In the Android app, even when I specifically set the wallabag-URL to http**s**://wallabag.mydomain.tld, it requests the non-https version from my server and can't connect. I solved this for be by adding a static redirect for http://wallabag.mydomain.tld to http**s**://wallabag.mydomain.tld, which breaks acme.sh compatibility but works for the android app. I know that I can just disable this redirect again after setting the app once, but it took me quite a while to figure this out and I think this is undesired behaviour.

So I think this issue needs to be addressed so that the Android app really uses http**s** when asked to.

Thank you!

**Comment by di72nn** [comment]
Hi,

Thanks for the detailed report.
I have the impression, that your reverse proxy is misconfigured (it was the reason for such problems a couple of times already). The thing is that the app prefers HTTPS by default, so the only way it can request HTTP is when it was redirected to a URL that uses HTTP protocol (that usually happens with the `/login` redirect). You can check [this](https://github.com/wallabag/android-app/issues/622#issuecomment-323596298).

**Comment by rklasen** [comment]
Hi,

thanks for your answer! I just tried `curl -I https://wallabag.mydomain.tld` and you're absolutely right, it redirects to `http://wallabag.mydomain.tld/login. So I guess we can close this issue, even though I don't understand why the redirect occurs. Apache only terminates the TLS connection and proxy passes to the docker container, so apparently the container sees no https connection and redirects to the non-https version. In the apache virtual host, I set the value `requestheader set x-forwarded-proto "https"`, but it doesn't seem to suffice.
